File: backend/app/models/course.py
from datetime import datetime
from enum import Enum

from sqlalchemy import DateTime, ForeignKey, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from backend.app.db.database import Base



class CourseStatus(str, Enum):
    DRAFT = 'draft'
    PUBLISHED = 'published'
    ARCHIVED = 'archived'


class Course(Base):
    __tablename__ = 'courses'

    id: Mapped[int] = mapped_column(primary_key=True)
    title: Mapped[str] = mapped_column(String(200), nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    teacher_id: Mapped[int] = mapped_column(ForeignKey('users.id'), nullable=False)
    status: Mapped[CourseStatus] = mapped_column(String(20), nullable=False, default=CourseStatus.DRAFT, server_default=CourseStatus.DRAFT.value)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)

    teacher = relationship('User')

File: frontend/src/types/course.ts
export type CourseStatus = 'draft' | 'published' | 'archived';


export type Course = {

    id: number;
    title: string;
    description: string | null;
    teacher_id: number;
    status: CourseStatus;
    created_at: string;

    };

export type CourseCreateRequest = {

    title: string;
    description?: string | null;


    };

export type CourseProgress = {
    course_id: number;
    total_lessons: number;
    completed_lessons: number;
    progress_percentage: number;

    };